dag_in_context/src/greedy_dag_extractor.rs

```diff
@@ -120,9 +120,12 @@ fn get_node_cost(
             .sum::<NotNan<f64>>();
 
     let mut costs: HashMap<ClassId, Cost> = Default::default();
-    for c in child_cost_sets.iter().map(|cs| &cs.costs) {
-        for (cid, cost) in c.iter() {
-            costs.insert(cid.clone(), *cost);
+    // children of region nodes are not shared
+    if !cm.regions.contains(op) {
```

Comment:
This doesn't seem right for loops- we still want to consider the inputs of the loops in the same region!

Reply:
Good catch! Should be fixed